lumeer-core/src/main/java/io/lumeer/core/facade/AbstractFacade.java

```diff
@@ -19,26 +19,37 @@
  */
 package io.lumeer.core.facade;
 
+import io.lumeer.api.model.Pagination;
 import io.lumeer.api.model.Permission;
 import io.lumeer.api.model.Permissions;
+import io.lumeer.api.model.Query;
 import io.lumeer.api.model.Resource;
 import io.lumeer.api.model.Role;
 import io.lumeer.core.AuthenticatedUser;
 import io.lumeer.core.PermissionsChecker;
+import io.lumeer.core.WorkspaceKeeper;
+import io.lumeer.core.cache.UserCache;
 import io.lumeer.core.model.SimplePermission;
+import io.lumeer.storage.api.query.SearchQuery;
 
 import java.util.Set;
 import javax.inject.Inject;
 
-public abstract class AbstractFacade {
+abstract class AbstractFacade {
+
+   @Inject
+   protected AuthenticatedUser authenticatedUser;
 
    @Inject
    protected PermissionsChecker permissionsChecker;
 
    @Inject
-   protected AuthenticatedUser authenticatedUser;
+   protected UserCache userCache;
 
-   public <T extends Resource> T keepOnlyActualUserRoles(final T resource) {
+   @Inject
+   protected WorkspaceKeeper workspaceKeeper;
+
+   protected <T extends Resource> T keepOnlyActualUserRoles(final T resource) {
       Set<Role> roles = permissionsChecker.getActualRoles(resource);
       Permission permission = new SimplePermission(authenticatedUser.getUserEmail(), roles);
 
```

Comment:
You often use `getCurrentUserName`

Reply:
It's basically the same. The only difference is that `getUserEmail()` just gets the email from Keycloak while `getCurrentUserName()` is a shortcut for `getUser().getUsername()` and `getUser()` tries to get user record from `UserCache`. It's mainly useful for getting the groups for a given user.